libraries/botbuilder/src/cloudAdapterBase.ts

```diff
@@ -0,0 +1,320 @@
+                    }
+                }
+
+                return { id: '' };
+            })
+        );
+    }
+
+    /**
+     * Update an activity
+     *
+     * @param {TurnContext} turnContext turn context
+     * @param {Activity} activity an activity to update
+     * @returns {Promise<ResourceResponse>} a promise resolving to a resource response
+     */
+    updateActivity(turnContext: TurnContext, activity: Partial<Activity>): Promise<ResourceResponse>;
+    async updateActivity(turnContext: unknown, activity: unknown): Promise<ResourceResponse> {
+        TurnContext.assert(turnContext, ['turnContext']);
+        assertPartialActivity(activity, ['activity']);
+
+        const connectorClient = this.turnContextConnectorClient(turnContext);
+
+        const conversationId = activity.conversation?.id;
+        assert.string(conversationId, ['activity', 'conversation', 'id']);
+
+        const activityId = activity.id;
+        assert.string(activityId, ['activity']);
+
+        return connectorClient.conversations.updateActivity(conversationId, activityId, activity);
+    }
+
+    /**
+     * Delete an activity
+     *
+     * @param {TurnContext} turnContext turn context
+     * @param {ConversationReference} reference a conversation reference to delete
+     * @returns {Promise<void>} a promise representing the async operation
+     */
+    deleteActivity(turnContext: TurnContext, reference: Partial<ConversationReference>): Promise<void>;
+    async deleteActivity(turnContext: unknown, reference: unknown): Promise<void> {
+        TurnContext.assert(turnContext, ['turnContext']);
+        assertPartialConversationReference(reference, ['reference']);
+
+        const connectorClient = this.turnContextConnectorClient(turnContext);
+
+        const conversationId = reference.conversation?.id;
+        assert.string(conversationId, ['reference', 'conversation', 'id']);
+
+        const activityId = reference.activityId;
+        assert.string(activityId, ['reference', 'activityId']);
+
+        await connectorClient.conversations.deleteActivity(reference.conversation.id, reference.activityId);
+    }
+
+    /**
+     * Continue a conversation
+     *
+     * @param {Partial<ConversationReference>} reference a conversation reference
+     * @param {BotLogic} logic the bot's logic
+     * @returns {Promise<void>} a promise representing the async operation
+     */
+    continueConversation(reference: Partial<ConversationReference>, logic: BotLogic): Promise<void>;
+    async continueConversation(reference: unknown, logic: unknown): Promise<void> {
+        assertPartialConversationReference(reference, ['reference']);
+        assertBotLogic(logic, ['logic']);
+
+        // TODO(jpg): this
+        const claimsIdentity = new ClaimsIdentity(
+            [
+                {
+                    type: AuthenticationConstants.AppIdClaim,
```

Comment:
I could use a pointer here - the Dotnet implementation has many overrides that obfuscate exactly what these claims should be.

Reply:
The value here should match the appId of the bot that is communicating in this continued conversation. To follow Semver, we ended up with a number of overloads in C#'s BotAdapter.

One scenario where an incorrect AppId causes problems is during OAuth flows, the AppId to be used is normally grabbed from the ClaimsIdentity in the TurnState.

Reply:
I think the only way we could do that is to add an `appId` parameter to the constructor then, as we don't have the necessary context elsewhere to extract the appId. Does that sound reasonable?